src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -916,7 +905,11 @@ private void getParameterEncryptionMetadata(Parameter[] params) throws SQLServer
 
 	/** Manage re-using cached handles */
 	private boolean reuseCachedHandle(boolean hasNewTypeDefinitions, boolean discardCurrentCacheItem) {
-		
+        if (definitionChanged || connection.contextChanged) {
+            prepStmtHandle = -1; // so that hasPreparedStatementHandle() also returns false
+            return false;
```

Comment:
we should also add these two in here, do you agree? 
```
connection.contextChanged = false;  // to be able to use the handle after this call
connection.clearCachedPreparedStatementHandle(); //making sure the cache is cleared
```

Reply:
Thank you so much for the suggestion, it's much better.